pkg/validator/controllers/generic.go

```diff
@@ -0,0 +1,126 @@
+package controllers
+
+import (
+	"time"
+
+	"github.com/fairwindsops/polaris/pkg/config"
+	"github.com/sirupsen/logrus"
+	kubeAPICoreV1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	kubeAPIMetaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+)
+
+// GenericController is a base implementation with some free methods for inherited structs
+type GenericController struct {
+	Name        string
+	Namespace   string
+	PodSpec     kubeAPICoreV1.PodSpec
+	ObjectMeta  kubeAPIMetaV1.ObjectMeta
+	Kind        config.SupportedController
+	KindString  string
+	CreatedTime time.Time
+}
+
+// GetPodSpec returns the original kubernetes template pod spec
+func (g GenericController) GetPodSpec() *kubeAPICoreV1.PodSpec {
+	return &g.PodSpec
+}
+
+// GetObjectMeta returns the metadata
+func (g GenericController) GetObjectMeta() kubeAPIMetaV1.ObjectMeta {
+	return g.ObjectMeta
+}
+
+// GetKind returns the supportedcontroller enum type
+func (g GenericController) GetKind() config.SupportedController {
+	return g.Kind
+}
+
+// GetKindString returns a string representing what kind of object the top level controller is.
+func (g GenericController) GetKindString() string {
+	if g.KindString == "" {
+		return g.Kind.String()
+	}
+	return g.KindString
+}
+
+// GetName is inherited by all controllers using generic controller to get the name of the controller
+func (g GenericController) GetName() string {
+	return g.Name
+}
+
+// GetNamespace is inherited by all controllers using generic controller to get the namespace of the controller
+func (g GenericController) GetNamespace() string {
+	return g.Namespace
+}
+
+// LoadControllers loads a list of controllers from the kubeResources Pods
+func LoadControllers(pods []kubeAPICoreV1.Pod, dynamicClientPointer *dynamic.Interface, restMapperPointer *meta.RESTMapper) []GenericController {
```

Comment:
Can we move this logic into `pkg/kube`? I'd like to keep all the API interactions in there.